### PR TITLE
Move all map items into their own files

### PR DESCRIPTION
**Pull request by DLi7077**
Came across the map file, and it looked kinda fat
Also, there's a bug where the entire map rerenders when we click on a restaurant marker because the selectedRestauarantId state is in the map component. It should've been in whichever component where we render a list of restaurants.
because of this bug, the map recenters to the avatar because the map component rerenders.

To fix this, we just move the selected restaurant state to a RestaurantMarkers component, with it's job to render all restaurants as well as the selected restaurant. The map (parent) doesn't need to know what the markers (children) have as the selected marker

Because of this requirement, we might as well move all of our existing components into their own files because that just seems to make more sense.

Map (parent) doesn't really care that our Avatar is using amongus png, and also doesn't care which restaurant is selected

TLDR: found bug, fix requires a lil refactor, refactor looks like cleaner architecture 

Before:

https://user-images.githubusercontent.com/51217000/236118374-a10ccba7-ac68-447c-b9f8-8bca357457f3.mp4

After:

https://user-images.githubusercontent.com/51217000/236118530-3be9c8aa-6487-468f-9741-dbab3750f814.mp4

### CODE REVIEW

**Review by Remos96 — APPROVED** [review]
Looks good, but nit: please provide PR description :)

**Comment by DLi7077** [comment]
> Looks good, but nit: please provide PR description :)

Eat my left foot
